src/parsers/manifest/dash/parse_adaptation_sets.ts

```diff
@@ -334,19 +343,11 @@ export default function parseAdaptationSets(
         isDub = true;
       }
 
-      const isClosedCaption = type === "text" &&
-                              accessibility != null &&
-                              isHardOfHearing(accessibility) ? true :
-                                                               undefined;
-      const isAudioDescription = type === "audio" &&
-                                 accessibility != null &&
-                                 isVisuallyImpaired(accessibility) ? true :
-                                                                     undefined;
-
-      const isSignInterpreted = type === "video" &&
-                                accessibility != null &&
-                                hasSignLanguageInterpretation(accessibility) ? true :
-                                                                               undefined;
+      const isClosedCaption = accessibilities?.some(isHardOfHearing);
```

Comment:
Just, now a non-text `Adaptation` could theorically be closed captions which is different than the previous code.
Maybe that makes sense though, I don't know the point of view of either DASH-IF or DVB-DASH here.

(When I wrote that checking `type` could be moved out of the loop, I still thought it may still be there, just outside the loop.)

What do you think?

Reply:
Yes, I see what you mean, it differs from the previous code now. I think it also serves a purpose if signalling what kind of media the accessibility refers to. So I re-add it, but outside of the some-function.